filters/include/pcl/filters/impl/filter_indices.hpp

```diff
@@ -74,7 +74,34 @@ pcl::removeNaNFromPointCloud (const pcl::PointCloud<PointT> &cloud_in,
   }
 }
 
+template<typename PointT> void
+pcl::FilterIndicesXYZ<PointT>::applyFilter (PointCloud &output)
+{
+  std::vector<int> indices;
+  if (keep_organized_)
+  {
+    bool temp = extract_removed_indices_;
+    extract_removed_indices_ = true;
+    applyFilter (indices);
+    extract_removed_indices_ = temp;
```

Comment:
I don't see the benefit of this. Technically `keep_organized_ = true` can force `extract_removed_indices_` to be true and vice-versa. Doing otherwise can result in indices being computed, yet the queries returning that they haven't (unless they access the indices regardless, which is clearly an undefined behavior from library PoV)

Reply:
I'm having trouble understanding you. 

> Technically `keep_organized_ = true` can force `extract_removed_indices_` to be true

Agreed. Currently, `keep_organized_ = true` forces `extract_removed_indices_` to be true in order to used the removed index list to invalidate the points that need to be filtered out.

> and vice-versa.

What exactly is the vice-versa behavior you're referring to?


> Doing otherwise can result in indices being computed, yet the queries returning that they haven't (unless they access the indices regardless, which is clearly an undefined behavior from library PoV)

Unclear what you meant here. 

If your point is "there's no reason in restoring `extract_removed_indices_` to it's previous value". That I agree. Ultimately `removed_indices_` will be filled with indexes despite what the user requested. We can always spew a warning.

Reply:
> What exactly is the vice-versa behavior you're referring to?

If `extract_removed_indices_` is false, `keep_organized_` is also false. Both variables are one and the same.

> Unclear what you meant here.

I thought there was an equivalent function for `extract_removed_indices_` as `getKeepOrganized ()` is for `keep_organized_`. That function returns indices vector, not the boolean.

> We can always spew a warning.

That'd be better. My point was that the implementation goes against what's asked (`extract_removed_indices = false` is ignored)

Reply:
> If `extract_removed_indices_` is false, `keep_organized_` has to be false

I would say that enforcing `keep_organized_` takes priority. Not respecting `extract_removed_indices_` introduces a small computation overhead while not respecting `keep_organized_` can completely disable certain algorithms in the next stage of your pipeline. 

> > We can always spew a warning.
> 
> That'd be better. My point was that the implementation goes against what's asked (`extract_removed_indices = false` is ignored)

:+1: Follow-up commit.